projects/igniteui-angular/migrations/common/UpdateChanges.ts

```diff
@@ -202,39 +210,40 @@ export class UpdateChanges {
                 continue;
             }
 
-            let base;
-            let replace;
+            let base: string;
+            let replace: string;
             let groups = 1;
             let searchPattern;
 
             if (type === BindingType.output) {
-                base = String.raw`\(${change.name}\)=(["'])`;
-                replace = `(${change.replaceWith})=$1`;
+                base = String.raw`\(${change.name}\)=(["'])(.*?)\1`;
+                replace = `(${change.replaceWith})=$1$2$1`;
             } else {
                 // Match both bound - [name] - and regular - name
-                base = String.raw`(\s\[?)${change.name}(\s*\]?=)(["'])`;
-                replace = String.raw`$1${change.replaceWith}$2$3`;
+                base = String.raw`(\s\[?)${change.name}(\s*\]?=)(["'])(.*?)\3`;
+                replace = String.raw`$1${change.replaceWith}$2$3$4$3`;
                 groups = 3;
             }
 
             let reg = new RegExp(base, 'g');
             if (change.remove || change.moveBetweenElementTags) {
                 // Group match (\1) as variable as it looks like octal escape (error in strict)
-                reg = new RegExp(String.raw`\s*${base}.*?${'\\' + groups}(?=\s|\>)`, 'g');
+                reg = new RegExp(String.raw`\s*${base}.*?(?=\s|\>)`, 'g');
                 replace = '';
             }
             switch (change.owner.type) {
-            case 'component':
-                searchPattern = String.raw`\<${change.owner.selector}[^\>]*\>`;
-                break;
-            case 'directive':
-                searchPattern = String.raw`\<[^\>]*[\s\[]${change.owner.selector}[^\>]*\>`;
-                break;
+                case 'component':
+                    searchPattern = String.raw`\<${change.owner.selector}[^\>]*\>`;
+                    break;
+                case 'directive':
+                    searchPattern = String.raw`\<[^\>]*[\s\[]${change.owner.selector}[^\>]*\>`;
+                    break;
             }
 
             const matches = fileContent.match(new RegExp(searchPattern, 'g'));
 
             for (const match of matches) {
+                let replaceStatement = replace;
```

Comment:
Just rename the actual variable TBH.

Reply:
The `base` value is declared only once outside of the loop's scope, so I reassign it here as to not mutate the `base` value